File: leetcode1658/leetcode1658.py
from collections import defaultdict

class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        target = sum(nums) - x
        if target < 0:
            return -1
        left = 0
        temp = 0
        ans = -float('inf')
        for right in range(len(nums)):
            temp += nums[right]
            while temp > target:
                temp -= nums[left]
                left += 1
            if temp == target:
                ans = max(ans, right - left + 1)
        return -1 if ans < 0 else len(nums) - ans

File: leetcode1658/leetcode1658.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int minOperations(vector<int>& nums, int x) {
            long long ans = INT_MIN, left = 0, sum = 0;
            for (auto num : nums) sum += num;
            long long target = sum - x;
            if (target < 0) return -1;
            long long temp = 0;
            for (int right = 0; right < nums.size(); right ++){
                temp += nums[right];
                while (temp > target)
                {
                    temp -= nums[left];
                    left ++;
                }
                if (temp == target){
                    ans = max(ans, right - left + 1);
                }
            }
            return ans == INT_MIN ? -1 : nums.size() - ans;
        }
    };